py/desiutil/dust.py

```diff
@@ -0,0 +1,384 @@
+# Copied on Nov/20/2016 from https://github.com/kbarbary/sfdmap/ commit: bacdbbd
+
+# Licensed under an MIT "Expat" license
+
+# Copyright (c) 2016 Kyle Barbary
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+desiutil.dust
+=============
+
+Get E(B-V) values from the Schlegel, Finkbeiner & Davis (1998) dust map.
+"""
+
+import os
+
+import numpy as np
+
+# require a FITS reader of some sort.
+try:
+    import fitsio
+    getdata = fitsio.read
+except ImportError:
+    try:
+        from astropy.io.fits import getdata
+    except ImportError:
+        raise ImportError("could not import fitsio or astropy.io.fits")
+
+__all__ = ['SFDMap', 'ebv']
+__version__ = "0.1.0"
+
+
+def _isiterable(obj):
+    """Returns `True` if the given object is iterable."""
+
+    try:
+        iter(obj)
+        return True
+    except TypeError:
+        return False
+
+# -----------------------------------------------------------------------------
+# Coordinate conversion
+#
+# astropy's coordinate conversions have a gigantic overhead of about
```

Comment:
Considering that this code was written several years ago, this assertion should be verified.  If it is no longer true, we should use Astropy.